Compiler error when using inexistent structs inside anonymous functions

**V version:** V 0.2.2 7f0c8d1.98c611d
**OS:** macOs

**What did you do?**
```
module main

struct Struc{
mut:
	f fn (s Another, i int) ?
}
```

**What did you expect to see?**
A nice error which says that Another struct does not exist (it works if i define the struct)
 
**What did you see instead?**
```
==================
/var/folders/66/bwj09mb97d3cpxbqdjhk40xm0000gp/T/v/repr2.78550558054493938.tmp.c:640:68: error: expected identifier
typedef Option_void (*anon_fn_placeholder_int__void)(main__Another,int);
                                                                   ^
1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.
```

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -55,6 +55,16 @@ pub fn (mut c Checker) struct_decl(mut node ast.StructDecl) {
 							field.type_pos)
 					}
 				}
+				field_sym := c.table.sym(field.typ)
+				if field_sym.kind == .function {
+					fn_info := field_sym.info as ast.FnType
+					c.ensure_type_exists(fn_info.func.return_type, fn_info.func.return_type_pos) or {
+						return
+					}
+					for param in fn_info.func.params {
+						c.ensure_type_exists(param.typ, param.type_pos) or { return }
+					}
+				}
 			}
 			if sym.kind == .struct_ {
 				info := sym.info as ast.Struct

--- a/vlib/v/checker/tests/unknown_type_in_anon_fn.out
+++ b/vlib/v/checker/tests/unknown_type_in_anon_fn.out
@@ -1,0 +1,7 @@
+vlib/v/checker/tests/unknown_type_in_anon_fn.vv:5:10: error: unknown type `Another`
+    3 | struct Struc{
+    4 | mut:
+    5 |     f fn (s Another, i int) ?
+      |             ~~~~~~~
+    6 | }
+    7 |

--- a/vlib/v/checker/tests/unknown_type_in_anon_fn.vv
+++ b/vlib/v/checker/tests/unknown_type_in_anon_fn.vv
@@ -1,0 +1,8 @@
+module main
+
+struct Struc{
+mut:
+	f fn (s Another, i int) ?
+}
+
+fn main() {}

